osconfig_tests/gcp_clients/gcp_clients.go

```diff
@@ -0,0 +1,32 @@
+package gcpclients
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/config"
+	"google.golang.org/api/option"
+)
+
+var (
+	storageClient  *storage.Client
+	osconfigClient *osconfig.Client
+)
+
+// GetStorageClient returns a singleton GCP client for osconfig tests
+func GetStorageClient(ctx context.Context) (*storage.Client, error) {
+	if storageClient != nil {
+		return storageClient, nil
+	}
+
+	return storage.NewClient(ctx, option.WithCredentialsFile(config.OauthPath()))
+}
+
+// GetOsConfigClient returns a singleton GCP client for osconfig tests
+func GetOsConfigClient(ctx context.Context) (*osconfig.Client, error) {
+	if osconfigClient != nil {
```

Comment:
These aren't thread safe. Will that ever be an issue? Will we ever run our tests in parallel from a single process?

Reply:
even today, the test suites are run on independent goroutines and each test case in a test suite is again run on independent goroutines.

Reply:
Then I guess the questions are: Should this be thread safe, it seems weird that it wouldn't be. What happens if two goroutines create different clients? Can the clients be created in init() or is double-check-locking or channels better?

Reply:
at first i thought this about the client being thread safe. Later, reading through multi-threading paradigms, i realized, it is about the initialization of the the client being thread safe.

yeah, you are right, the client initialization of the singleton client is not thread safe. Double check locking or channels are great way to handle singleton initialization. However, I figured a more idiomatic go way to handle this. The go `sync` library provides a a type called `Once`.
More details [sync.Once](https://golang.org/pkg/sync/#Once)